component-tools/src/main/java/org/talend/sdk/component/tools/ComponentValidator.java

```diff
@@ -801,6 +801,12 @@ private boolean isListString(final Field f) {
                 && pt.getActualTypeArguments().length == 1 && pt.getActualTypeArguments()[0] == String.class;
     }
 
+    private boolean isListObject(final Field f) {
```

Comment:
you could directly test for collection :  

```java
   public boolean isCollectionOf(final Field f, Class<?> typeArg) {
        final ParameterizedType pt = ParameterizedType.class.cast(f.getGenericType());
        Type classType = pt.getRawType();
        if (pt.getActualTypeArguments().length != 1) {
            return false;
        }
        Type innerType = pt.getActualTypeArguments()[0];
        return classType instanceof Class<?> 
           && Collection.class.isAssignableFrom((Class<?>) classType) 
           && innerType instanceof Class<?> 
           && ((Class<?>) innerType).isAssignableFrom(typeArg);
    }
```

With this; a declared field `ArrayList<String>` function return true for  `isCollectionOf(field, String.class);`  but false for a field  `List<Object>`
with field `List<Object>`, function return true for `isCollectionOf(field, String.class);` (you can add a String to a List<Object> collection).

Reply:
Good point Chris.